src/Symfony/Component/DependencyInjection/Attribute/AutowireCallable.php

```diff
@@ -45,7 +45,7 @@ public function __construct(
 
     public function buildDefinition(mixed $value, ?string $type, \ReflectionParameter $parameter): Definition
     {
-        return (new Definition($type = \is_string($this->lazy) ? $this->lazy : ($type ?: 'Closure')))
+        return (new Definition($type = $type ?: 'Closure'))
```

Comment:
the goal is to take the type set in the lazy property when there is one, so we should handle the case of an array instead.
`current($this->lazy)` should do it

Reply:
So something like that 

```php
return (new Definition($type = \is_array($this->lazy) ? current($this->lazy) : ($type ?: 'Closure')))
```
Should I target 5.4 instead 7.2 ?

Reply:
5.4 if the but is there

Reply:
I rebased on 6.4